File: Front/WebShop-Frontend/src/app/models/commentRequest.ts
import { User } from "./user";

export class CommentRequest {
    content: string;
    productId: number;
    userId: number;

    constructor(content: string, productId: number, userId:number) {
        this.content = content;
        this.productId = productId;
        this.userId = userId;
    }

}

File: Back/WebShop-Backend/src/main/java/com/example/WebShop/models/requests/CommentRequest.java
package com.example.WebShop.models.requests;

import com.example.WebShop.models.entities.ProductEntity;
import com.example.WebShop.models.entities.UserEntity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.NonNull;

@Data
public class CommentRequest {
    @NotBlank
    private String content;
    @NonNull
    private Integer productId;
    @NonNull
    private Integer userId;
}
